src/streaming/utils/CapabilitiesFilter.js

```diff
@@ -172,19 +172,21 @@ function CapabilitiesFilter() {
         const configurations = [];
 
         manifest.Period.forEach((period) => {
-            period.AdaptationSet.forEach((as) => {
-                if (adapter.getIsTypeOf(as, type)) {
-                    as.Representation.forEach((rep, i) => {
-                        const codec = adapter.getCodec(as, i, false);
-                        _processCodecToCheck(type, rep, codec, configurationsSet, configurations);
-
-                        const supplementalCodecs = adapter.getSupplementalCodecs(rep)
-                        if (supplementalCodecs.length > 0) {
-                            _processCodecToCheck(type, rep, supplementalCodecs[0], configurationsSet, configurations);
-                        }
-                    });
-                }
-            });
+            if (!period.ImportedMPD) {
```

Comment:
This raises an interesting use case: If the Period element that contains `ImportedMPD` has default content, it might need to get filtered. However, the content will change if the referenced MPD is valid and can be merged.

Reply:
You're right. We need to handle the scenario where a Period contains both "regular content" and an ImportedMPD. In such cases, the regular content might need to be filtered out if the referenced MPD is valid and can be merged.